src/Components/Activities/ActivitiesForm.js

```diff
@@ -1,48 +1,202 @@
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import TextField from '@mui/material/TextField';
+import Card from '@mui/material/Card';
+import Button from '@mui/material/Button';
+import CardHeader from '@mui/material/CardHeader';
+import { Alert } from '@mui/material';
+import { CKEditor } from '@ckeditor/ckeditor5-react';
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
+import { useDropzone } from 'react-dropzone';
+import Swal from 'sweetalert2';
 import '../FormStyles.css';
+import '../../Styles/ActivitiesForm.css';
 
-const ActivitiesForm = () => {
+const thumb = {
+  display: 'inline-flex',
+  borderRadius: 2,
+  border: '1px solid #eaeaea',
+  margin: 'auto',
+  width: 'auto',
+  height: 100,
+  padding: 4,
+  boxSizing: 'border-box',
+};
+
+const ActivitiesForm = ({ id }) => {
   const [initialValues, setInitialValues] = useState({
-    name: '',
-    description: '',
+    name: id ? id.data.name : '',
+    description: id ? id.data.description : '',
+    image: id ? id.data.name : null,
   });
 
-  const handleChange = (e) => {
-    if (e.target.name === 'name') {
-      setInitialValues({ ...initialValues, name: e.target.value });
-    }
-    if (e.target.name === 'description') {
-      setInitialValues({ ...initialValues, description: e.target.value });
-    }
-  };
+  const [files, setFiles] = useState([]);
+  const { getRootProps, getInputProps } = useDropzone({
+    multiple: false,
+    maxFiles: 1,
+    accept: 'image/jpeg, image/png',
+    onDrop: (acceptedFiles) => {
+      setInitialValues({
+        ...initialValues,
+        image: acceptedFiles.map((file) =>
+          Object.assign(file, {
+            preview: URL.createObjectURL(file),
+          }),
+        ),
+      });
+
+      setFiles(
+        acceptedFiles.map((file) =>
+          Object.assign(file, {
+            preview: URL.createObjectURL(file),
+          }),
+        ),
+      );
+    },
+  });
 
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    console.log(initialValues);
+  const thumbs = files.map((file) => (
+    <div key={file.name} style={thumb}>
+      <div className="thumb-inner">
+        <img className="img" src={file.preview} />
+      </div>
+    </div>
+  ));
+
+  useEffect(
+    () => () => {
+      // Make sure to revoke the data uris to avoid memory leaks
+      files.forEach((file) => URL.revokeObjectURL(file.preview));
+    },
+    [files],
+  );
+
+  const handleClick = (values) => {
+    let data = {
+      name: values.name,
+      description: values.description,
+      image: values.image,
+    };
+
+    Swal.fire('Good job!', 'You clicked the button!', 'success');
+    if (id) {
+      Axios.patch(`http://ongapi.alkemy.org/api/categories/${id}`, data).then(
+        () => {
+          Swal.fire('success');
+        },
+      );
+    } else {
+      Axios.post('http://ongapi.alkemy.org/api/categories/', data).then(() => {
+        Swal.fire('success');
+      });
+    }
```

Comment:
move this axios call to the activitiesService. 
- Create a constant with the URL
- Create a createActivity functionallity
- Create a updateActivity functionallity
- Create a createOrUpdateActivity functionallity

Then call the createOrUpdate here
```
createOrUpdate(id).then(()=> Swal.fire('success'))
```